### PR TITLE
UIP-749

### PR DESCRIPTION
**Pull request by emilysimonton**

### CODE REVIEW

File: src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+public class UnifiedFsSourceTask extends FsSourceTask {
+    private static final Logger log = LoggerFactory.getLogger(FsSourceTask.class);
+
+    private AtomicBoolean      stop;
+    private FsSourceTaskConfig config;
+    private Policy             policy;
+
+    @Override
+    public void start(Map<String, String> properties) {
```
Comment (emilysimonton) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java]: tried to use start from FsSourceTask however stop, config and policy are all private variables so I was unable to access them. I could either change those variables to protected in FsSourceTask, or override start in UnifiedSourceTask

File: src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        try {
+            return asStream(policy.execute())
+                .filter(metadata -> metadata.getLen() > 0)
+                .collect(Collectors.toList());
+        } catch (IOException | ConnectException e) {
+            //when an exception happens executing the policy, the connector continues
+            log.error("Cannot retrive files to process from FS: " + policy.getURIs() + ". Keep going...", e);
+            return Collections.EMPTY_LIST;
+        }
+    }
```
Comment (emilysimonton) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java]: filesToProcess() above and asStream() below are both private which is why I brought them into UnifiedFsSourceTask. Again, I could make these protected, or I could keep it as is.

**Comment by emilysimonton** [comment]
I still need to work on adding/updating unit tests, but opening up a pr to get eyes on it.

File: src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceConnectorConfig.java [hunk 3]
```diff
@@ -17,6 +17,9 @@
     public static final String TOPIC = "topic";
     private static final String TOPIC_DOC = "Topic to copy data to.";
 
+    public static final String REPLAY_FILE = "policy.fs.replayFile";
+    private static final String REPLAY_FILE_DOC = "T/F depending on if we are replaying a file.";
```
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceConnectorConfig.java]: I keep doing a double take when reading this config. Lets rename it to be more specific about it's behavior. Maybe `ignoreFileOffset`, `replayAll`, `fromBeginning`?

File: src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java (hunk 1 above)
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java]: You mean the class members right?
```
private AtomicBoolean      stop;
private FsSourceTaskConfig config;
private Policy             policy
```
Let's just make them protected in the parent class, so you don't have to do all this code duplication.

File: src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java (hunk 2 above)
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java]: Would making the class members protected still require exposing these methods?

File: src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
+package com.github.mmolimar.kafka.connect.fs.unified;
```
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/unified/UnifiedFsSourceTask.java]: Can we try to put these into `com.unified.platform.connect.fs`? This is to be consistent in our packaging if possible. Not sure if there's any particular package scope we might be missing when running or testing. If it doesn't cause us any problems, then lets move it.

File: src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java [hunk 5]
```diff
@@ -98,24 +98,29 @@ public void start(Map<String, String> properties) {
         }
     }
 
-    private <T> Stream<T> asStream(Iterator<T> src) {
+    protected <T> Stream<T> asStream(Iterator<T> src) {
```
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java]: do these two functions need to be exposed? Are you using them in your subclass?

File: src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java [hunk 6]
```diff
@@ -22,11 +22,11 @@
 import java.util.stream.StreamSupport;
 
 public class FsSourceTask extends SourceTask {
-    private static final Logger log = LoggerFactory.getLogger(FsSourceTask.class);
+    protected static final Logger log = LoggerFactory.getLogger(FsSourceTask.class);
```
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java]: Definitely keep this private! We should create our own logger in the subclass or else we'll be logging out with the wrong class name. ;)

File: src/main/java/com/github/mmolimar/kafka/connect/fs/UnifiedFsSourceTask.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import org.apache.kafka.connect.source.SourceRecord;
+
+import com.github.mmolimar.kafka.connect.fs.file.FileMetadata;
+import com.github.mmolimar.kafka.connect.fs.file.reader.FileReader;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+
+public class UnifiedFsSourceTask extends FsSourceTask {
+
```
Comment (wiyu) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/UnifiedFsSourceTask.java]: Create our own logger here.

**Pull request merged**